tapscript/script.go

```diff
@@ -48,3 +52,87 @@ func NewChannelFundingScriptTree() *FundingScriptTree {
 		},
 	}
 }
+
+// NewChannelFundingScriptTreeUniqueID creates a new funding script tree for a
+// custom channel asset-level script key. The script tree is constructed with a
+// simple OP_TRUE script that allows anyone to spend the output and another
+// OP_RETURN script that makes the resulting script key unique. This simplifies
+// the funding process as no signatures for the asset-level witnesses need to be
+// exchanged. This is still safe because the BTC level multi-sig output is still
+// protected by a 2-of-2 MuSig2 output.
+func NewChannelFundingScriptTreeUniqueID(id asset.ID) (*FundingScriptTree,
```

Comment:
Do we also need to worry about the case where multiple asset UTXOs w/ the same asset ID exist in the funding output? Or is the assumption that the funder won't create such an output, as they can make a "smaller" one with the UTXO value consolidated (as tehy're the same assetID after all).

Reply:
Exactly, the same assetIDs would always be merged into a single UTXO (see comment above as well).